js/components/SigninForm/SigninForm.js

```diff
@@ -25,9 +28,13 @@ export default class SigninForm extends Component {
     return errors
   }
   render() {
+    const { navigation } = this.props
     return (
       <Form
-        onSubmit={values => this.onSubmit(values)}
+        onSubmit={values => {
+          this.onSubmit(values)
+          navigation.navigate('App')
+        }}
         validate={values => this.validate(values)}
         render={({ handleSubmit, pristine, invalid }) => (
```

Comment:
use pristine to inactivate your submit button when the user has not filled out the whole form.